sources/Renderer/OpenGL/Platform/Linux/LinuxGLContextX11.cpp

```diff
@@ -0,0 +1,369 @@
+/*
+ * LinuxGLContextX11.cpp
+ *
+ * Copyright (c) 2015 Lukas Hermanns. All rights reserved.
+ * Licensed under the terms of the BSD 3-Clause license (see LICENSE.txt).
+ */
+
+#include "LinuxGLContextX11.h"
+#include "../../Ext/GLExtensions.h"
+#include "../../Ext/GLExtensionLoader.h"
+#include "../../GLCore.h"
+#include "../../../CheckedCast.h"
+#include "../../../StaticAssertions.h"
+#include "../../../RenderSystemUtils.h"
+#include "../../../../Core/CoreUtils.h"
+#include "../../../../Core/Assertion.h"
+#include "../../../../Platform/Linux/LinuxDisplay.h"
+#include <LLGL/Backend/OpenGL/NativeHandle.h>
+#include <LLGL/Log.h>
+
+namespace LLGL
+{
+
+
+#ifndef GLX_CONTEXT_MAJOR_VERSION_ARB
+#define GLX_CONTEXT_MAJOR_VERSION_ARB 0x2091
+#endif
+
+#ifndef GLX_CONTEXT_MINOR_VERSION_ARB
+#define GLX_CONTEXT_MINOR_VERSION_ARB 0x2092
+#endif
+
+typedef GLXContext (*GXLCREATECONTEXTATTRIBARBPROC)(::Display*, GLXFBConfig, GLXContext, Bool, const int*);
+
+
+/*
+ * LinuxGLContextX11 class
+ */
+
+LinuxGLContextX11::LinuxGLContextX11(
+    const GLPixelFormat&                    pixelFormat,
+    const RendererConfigurationOpenGL&      profile,
+    Surface&                                surface,
+    LinuxGLContextX11*                         sharedContext,
+    const OpenGL::RenderSystemNativeHandle* customNativeHandle)
+:
+    samples_ { pixelFormat.samples }
+{
+    /* Notify the shared X11 display that it'll be used by libGL.so to ensure a clean teardown */
+    LinuxSharedX11Display::RetainLibGL();
+
+    /* Create GLX or proxy context if a custom one is specified */
+    NativeHandle nativeWindowHandle = {};
+    surface.GetNativeHandle(&nativeWindowHandle, sizeof(nativeWindowHandle));
+
+    if (customNativeHandle != nullptr)
+    {
+        CreateProxyGLXContext(pixelFormat, nativeWindowHandle, *customNativeHandle);
+    }
+    else
+    {
+        CreateGLXContext(pixelFormat, profile, nativeWindowHandle, sharedContext);
+    }
+}
+
+LinuxGLContextX11::~LinuxGLContextX11()
+{
+    if (!isProxyGLC_) {
```

Comment:
It looks like this implementation has not been copied over into a new class 1:1. Besides the different formatting style, `isProxyGLC_` is not set to true in the upper branch that calls `CreateProxyGLXContext()` anymore. Ideally, this file should be almost identical to what the old `LinuxGLContext.cpp looked like, except for the `GLContext::Create()` implementation of course. Can you please revise this file by simply copying the old LinuxGLContext.cpp file and remove the parts that have to stay in the new version of that file? FWIW: I assume git might even recognize this file as being moved and modified in that case.